### PR TITLE
[Concept Entry] C# Interfaces

### PR DESCRIPTION
**Pull request by manviii27**
### Description

 Added a new concept in C# `Interfaces` . File Location: `docs/content/c-sharp/concepts/interfaces/interfaces.md`

### Type of Change

- Adding a new entry

### Checklist

- [ ✓] All writings are my own.
- [ ✓] My entry follows the Codecademy Docs style guide.
- [ ✓] My changes generate no new warnings.
- [ ✓] I have performed a self-review of my own writing and code.
- [ ✓] I have checked my entry and corrected any misspellings.
- [ ✓] I have made corresponding changes to the documentation if needed.
- [ ✓] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ ✓] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ ✓] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/c-sharp/concepts/interfaces/interfaces.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+CatalogContent:
+  - 'learn-c-sharp'
+  - 'paths/computer-science'
+---
+
+An **interface** in C# is a contract that defines a set of methods, properties, events, and indexers that a class or struct must implement. Interfaces cannot be instantiated directly, but they can be implemented by classes and structs. They are used to achieve abstraction and polymorphism in C#.
+
+### Syntax
+
+To create an interface in C#, you use the `interface` keyword. The syntax for defining an interface in C# is:
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/interfaces/interfaces.md:21]:
```suggestion
An interface in C# is created using the `interface` keyword. The syntax for defining an interface in C# is:
```
No first- or second-person pronouns in Docs.

File: content/c-sharp/concepts/interfaces/interfaces.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        set { }
+    }
+
+    public event EventHandler MyEvent;
+}
+```
+
+### Example
+
+The following examples show various comment styles:
```
Comment (SSwiniarski) [inline content/c-sharp/concepts/interfaces/interfaces.md:54]:
```suggestion
The following example shows implementation of an interface:

```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@manviii27 I did a first review and only had a couple of comments.

**Comment by manviii27** [comment]
@SSwiniarski I've made the changes, please check. Thank you.

**Review by SSwiniarski — APPROVED** [review]
@manviii27 this is ready for a second review.

**Review by caupolicandiaz — APPROVED** [review]
@manviii27 @SSwiniarski Just made a couple minor edits. Looks good to merge

**Pull request merged**